pkg/commands/plan/plan_init.go

```diff
@@ -0,0 +1,243 @@
+	c.githubClient = github.NewClient(
+		ctx,
+		c.GitHubFlags.FlagGitHubToken,
+		github.WithInitialRetryDelay(c.RetryFlags.FlagInitialRetryDelay),
+		github.WithMaxRetries(c.RetryFlags.FlagMaxRetries),
+		github.WithMaxRetryDelay(c.RetryFlags.FlagMaxRetryDelay),
+	)
+
+	if len(c.flagDirectories) == 0 {
+		c.flagDirectories = append(c.flagDirectories, cwd)
+	}
+
+	logger.Debugln("finding entrypoint directories..")
+
+	dirs, err := terraform.GetEntrypointDirectories(c.flagDirectories)
+	if err != nil {
+		return fmt.Errorf("failed to find terraform directories: %w", err)
+	}
+
+	c.entrypoints = dirs
+	logger.Debugf("terraform entrypoint directories: %s", c.entrypoints)
+
+	return c.Process(ctx)
+}
+
+// Process handles the main logic for the Guardian plan init process.
+func (c *PlanInitCommand) Process(ctx context.Context) error {
+	logger := logging.FromContext(ctx)
+	logger.Debugln("starting Guardian plan init")
+
+	if c.GitHubFlags.FlagGitHubAction && c.flagDeleteOutdatedComments {
+		logger.Debugln("removing outdated comments...")
+		if err := c.deleteOutdatedComments(ctx, c.GitHubFlags.FlagGitHubOwner, c.GitHubFlags.FlagGitHubRepo, c.flagPullRequestNumber); err != nil {
+			return fmt.Errorf("failed to delete outdated comments: %w", err)
+		}
+	}
+
+	diffDirs, err := c.gitClient.DiffDirsAbs(ctx, c.flagDestRef, c.flagSourceRef)
+	if err != nil {
+		return fmt.Errorf("failed to find git diff directories: %w", err)
+	}
+	logger.Debugf("changed directories: %+v", diffDirs)
+
+	targetDirs := util.GetSliceIntersection(c.entrypoints, diffDirs)
+	logger.Debugf("target directories: %+v", targetDirs)
+
+	// convert to child path for output, using absolute path
+	// creates an ugly github workflow name
+	for k, dir := range targetDirs {
+		childPath, err := util.ChildPath(c.workingDir, dir)
+		if err != nil {
+			return fmt.Errorf("failed to get child path for: %w", err)
+		}
+		targetDirs[k] = childPath
+	}
+
+	if c.flagJSON {
+		outJSON, err := json.Marshal(targetDirs)
+		if err != nil {
+			return fmt.Errorf("failed to create json string: %w", err)
+		}
+		c.Outf(string(outJSON))
+	} else {
+		for _, dir := range targetDirs {
+			c.Outf("%s", dir)
+		}
+	}
+
+	return nil
+}
+
+// deleteOutdatedComments deletes the pull request comments from previous Guardian plan runs.
+func (c *PlanInitCommand) deleteOutdatedComments(ctx context.Context, owner, repo string, number int) error {
+	listOpts := &gh.IssueListCommentsOptions{
+		ListOptions: gh.ListOptions{PerPage: 100},
+	}
+
+	for {
+		comments, paging, err := c.githubClient.ListIssueComments(ctx, owner, repo, number, listOpts)
+		if err != nil {
+			return fmt.Errorf("failed to list comments: %w", err)
+		}
+
+		for _, comment := range comments {
+			if strings.HasPrefix(comment.Body, planCommentPrefix) {
+				if err := c.githubClient.DeleteIssueComment(ctx, owner, repo, comment.ID); err != nil {
+					return fmt.Errorf("failed to delete comment: %w", err)
+				}
+			}
```

Comment:
Style nit, usually we try to avoid this level of deep nesting:

```suggestion
			if !strings.HasPrefix(comment.Body, planCommentPrefix) {
				continue
			}

			if err := c.githubClient.DeleteIssueComment(ctx, owner, repo, comment.ID); err != nil {
				return fmt.Errorf("failed to delete comment: %w", err)
			}
```